### PR TITLE
Make prim_eval work

### PR DESCRIPTION
**Pull request by mat-hek**
- Copy prim_eval.S to FissionLib - no idea how to make it work otherwise, but didn't dig
- Mock erlang:bump_reductions - see https://linear.app/swmansion/project/erlangbump-reductions-missing-02f6ff09cac1/overview
- Mock erl_anno:set_file - see https://linear.app/swmansion/project/erl-annoset-file2-sometimes-crashes-in-erl-eval-76e146323242/overview
- Add tests 🤯

### CODE REVIEW

File: fission_lib/lib/fission_lib.ex [hunk 1]
```diff
@@ -15,24 +15,25 @@ defmodule FissionLib do
...
       )
       |> Map.new()
 
     :packbeam_api.create(
       ~c"#{options.output_path}",
-      Enum.map(
-        [options.fission_lib_path] ++ Path.wildcard("#{@build_path}/**/*.{beam,app}"),
-        &String.to_charlist/1
-      ),
+      Enum.map([options.fission_lib_path] ++ options.artifacts, &String.to_charlist/1),
```
Comment (jgonet) [inline fission_lib/lib/fission_lib.ex]: Nit: `[options.fission_lib_path | options.artifacts]`

File: fission_lib/patches/estdlib/erl_anno.erl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+%% Patch reason: original set_file/2 crashes sometimes ¯\_(ツ)_/¯
+%% for example in "spawn" test in eval_test.exs
+%% See https://linear.app/swmansion/project/erl-annoset-file2-sometimes-crashes-in-erl-eval-76e146323242/overview
+
+-module(erl_anno).
+
+-export([set_file/2]).
+
+set_file(_File, Anno) -> Anno.
```
Comment (jgonet) [inline fission_lib/patches/estdlib/erl_anno.erl:9]: We may be getting to last branch – filename isn't list or binary thus not settable.

```erlang
set_file(File, Anno) ->
    set(file, File, Anno).

set(Item, Value, Anno) ->
    case {is_settable(Item, Value), Anno} of
        {true, Line} when ?ALINE(Line) ->
            set_anno(Item, Value, [{location, Line}]);
        {true, {L, C} = Location} when ?ALINE(L), ?ACOLUMN(C) ->
            set_anno(Item, Value, [{location, Location}]);
        {true, A} when is_list(A), A =/= [] ->
            set_anno(Item, Value, Anno);
        _ ->
            erlang:error(badarg, [Item, Value, Anno])
    end.
```

File: fission_lib/config/config.exs [hunk 3]
```diff
@@ -1,3 +1,9 @@
 import Config
 
 config :fission_lib, add_tracing: true
```
Comment (TheSobkiewicz) [inline fission_lib/config/config.exs:3]: Add atomvm_path with default value to make it easier to track.

Comment (mat-hek) [reply fission_lib/config/config.exs:3]: I don't think there's a default value for the atomvm path... I added a comment a few lines below

File: fission_lib/patches/estdlib/erl_anno.erl (hunk 2 above)
Comment (mat-hek) [reply fission_lib/patches/estdlib/erl_anno.erl:9]: Looked at this for a while, not sure why it happens. I'll merge this and we can tackle it in a separate PR

**Pull request merged**